Minor api improvements

lehins opened the pull request:
# Description

* Rename TxId field from _unTxId to unTxId
* Some TxAuxData realeted cleanups:
  * Remove redundant pattern synonym `AllegraTxAuxData'` and `AlonzoTxAuxData'`
  * Hide internal `AllegraTxAuxDataRaw` constructor with `atadrMetadata` and `atadrTimelock`
    record fields.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [ ] ~New tests are added if needed and existing tests are updated~
- [x] Any changes are noted in the `CHANGELOG.md` for affected package
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

lehins committed:
Export `Tx` and `mkTxIx`

lehins committed:
Rename `TxId` field from `_unTxId` to `unTxId`

lehins committed:
Remove redundant pattern synonym `AlonzoTxAuxData'`

lehins committed:
Remove redundant constraint from `AllegraTxAuxData` encoder

lehins committed:
Remove redundant pattern synonym `AllegraTxAuxData'`

lehins committed:
Hide internal `AllegraTxAuxDataRaw` constructor

JaredCorduan commented on libs/cardano-ledger-core/src/Cardano/Ledger/TxIn.hs:
```diff
@@ -52,7 +52,7 @@ import NoThunks.Class (NoThunks (..))
 -- ====================================================================================
 
 -- | A unique ID of a transaction, which is computable from the transaction.
-newtype TxId c = TxId {_unTxId :: SafeHash c EraIndependentTxBody}
```
given that this one might be used by sever downstream libraries, would it makes sense to make a standalone `_unTxId` function with the same semantics, and immediately deprecate it?

lehins replied:
We can't really create a function with exact semantics as a record field has. In other words a function cannot be used to for updates or construction, eg:
```haskell
let txId = TxId { _unTxId = "...."}
```

However, adding at least a deprecated function `_unTxId` will not hurt. Done.

JaredCorduan replied:
of course, that makes sense, thanks! what you added is exactly what I had in mind.

JaredCorduan reviewed (approved):
LGTM!

lehins committed:
Add deprecated `_unTxId` function

lehins committed:
Fix Conway CDDL test for Coin

Starting with Conway we no longer allow zero value. However `Coin`,
whenever used in other contexts, eg. fees, params, etc. are allowed to
be zero. Therefore we added a new CDDL type `positive_coin`, which is used
in `Value` and consequently in `TxOut`s. This commit fixes the test
only, implementation is totally correct
